plugins/network-generator/src/main/kotlin/dev/icerock/moko/network/MultiPlatformNetworkGeneratorPlugin.kt

```diff
@@ -16,28 +16,54 @@ class MultiPlatformNetworkGeneratorPlugin : Plugin<Project> {
     private val openApiGenerator = OpenApiGeneratorPlugin()
 
     override fun apply(target: Project) {
-        openApiGenerator.apply(target)
 
-        val generatedDir = "${target.buildDir}/generate-resources/main"
+        val mokoNetworkExtension = target.extensions.create("mokoNetwork", SpecConfig::class.java)
 
         target.afterEvaluate {
-            extensions.findByType(KotlinMultiplatformExtension::class.java)?.run {
-                val sourceSet = sourceSets.getByName(KotlinSourceSet.COMMON_MAIN_SOURCE_SET_NAME)
-                val sources = "$generatedDir/src/main/kotlin"
-                sourceSet.kotlin.srcDir(sources)
-            }
 
-            val removeGeneratedCodeTask =
-                tasks.create("removeGeneratedOpenApiCode", Delete::class) {
-                    delete(file(generatedDir))
+            mokoNetworkExtension.specs.forEach { spec ->
+
+                val generatedDir =
+                    "${target.buildDir}/generate-resources/main/src/main/kotlin/${spec.name}"
+
+                extensions.findByType(KotlinMultiplatformExtension::class.java)?.run {
```

Comment:
`extensions.findByType(KotlinMultiplatformExtension::class.java)` better do before forEach:
```
val multiplatformExtension = extensions.findByType(KotlinMultiplatformExtension::class.java)
mokoNetworkExtension.specs.forEach { spec ->
    ...
}
```

Reply:
and getting of sourceSet too